### PR TITLE
Stablehlo reduction dims fix

### PR DESCRIPTION
**Pull request by LPanosTT**
Allowing all reduction dims from `stablehlo.reduce` to materialize in `ttir` reduction ops

FYI: @AleksKnezevic

### CODE REVIEW

File: test/ttmlir/Conversion/StableHLOToTTIR/reduce/reduce_multi_dim.mlir [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// REQUIRES: stablehlo
+// RUN: ttmlir-opt --stablehlo-to-ttir-pipeline %s | FileCheck %s
+module @jit_reduce_add attributes {} {
+  func.func public @test_reduce_add(%arg0: tensor<1x64x112x112xf32>) -> tensor<1x64xf32>{
+    %cst_0 = stablehlo.constant dense<0.000000e+00> : tensor<f32>
+    %0 = stablehlo.reduce(%arg0 init: %cst_0) applies stablehlo.add across dimensions = [2, 3] : (tensor<1x64x112x112xf32>, tensor<f32>) -> tensor<1x64xf32>
+    // CHECK: %[[C:.*]] = tensor.empty[[C:.*]]
+    // CHECK: %[[C:.*]] = "ttir.sum"[[C:.*]]
```
Comment (mmanzoorTT) [inline test/ttmlir/Conversion/StableHLOToTTIR/reduce/reduce_multi_dim.mlir:8]: Please add a `CHECK` statement to verify that both `dims` are added.

File: test/ttmlir/Conversion/StableHLOToTTIR/reduce/reduce_multi_dim.mlir [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    return %0 : tensor<1x64xf32>
+  }
+}
```
Comment (mmanzoorTT) [inline test/ttmlir/Conversion/StableHLOToTTIR/reduce/reduce_multi_dim.mlir:11]: I'll suggest to add another function to tests more than 2 `dims`.

**Review by mmanzoorTT — APPROVED** [review]

**Comment by mrakitaTT** [comment]
@LPanosTT Now I am starting to get suspicious that you are secretly tracking my branches and doing this on purpose :smile: Just kidding of course, but this is third time we are fixing the same thing xD Since I've already covered this [here](https://github.com/tenstorrent/tt-mlir/pull/1625) as part of the broader reduce ops fixing and added different tests, do you mind closing this PR so I don't have to resolve conflicts? 🙏

**Comment by LPanosTT** [comment]
> @LPanosTT Now I am starting to get suspicious that you are secretly tracking my branches and doing this on purpose 😄 Just kidding of course, but this is third time we are fixing the same thing xD Since I've already covered this [here](https://github.com/tenstorrent/tt-mlir/pull/1625) as part of the broader reduce ops fixing and added different tests, do you mind closing this PR so I don't have to resolve conflicts? 🙏

@mrakitaTT oh dear my fault. I must’ve read past that part in your PR description. Thank a ton!

**Comment by mrakitaTT** [comment]
@LPanosTT no worries, I've sent it after yours, it was just a wild coincidence 😅 Thanks!

**Pull request closed without merging**